tools/ci_build/github/azure-pipelines/templates/win-x86-ci.yml

```diff
@@ -54,23 +54,25 @@ jobs:
           arguments: '$(BuildCommand) --test --config Debug'
           workingFolder: '$(Build.BinariesDirectory)'
 
-      - task: VSBuild@1
+      - task: MSBuild@1
         displayName: 'Build C# Debug'
         inputs:
-          solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.sln'
-          platform: 'any cpu'
+          solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.proj'
+          platform: 'Any CPU'
           configuration: 'Debug'
-          restoreNugetPackages: false
-          msbuildArchitecture: 'x86'
-          workingFolder: '$(Build.SourcesDirectory)\csharp'
+          msbuildArchitecture: 'x64'
           msbuildArguments: '/m'
+          workingFolder: '$(Build.SourcesDirectory)\csharp'
 
-      - task: CmdLine@2
+      - task: MSBuild@1
         displayName: 'Test C# Debug'
         inputs:
-          script: |
-           FOR /F  %%f IN ('dir /s /b test\Microsoft.ML.OnnxRuntime.Tests\bin\*Tests.dll') DO %dotnetexe% vstest %%f 
-          workingDirectory: '$(Build.SourcesDirectory)\csharp'
+          solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.proj'
+          platform: 'Any CPU'
+          configuration: 'Debug'
+          msbuildArchitecture: 'x64'
```

Comment:
Are the unit tests run in a 32-bit process or 64-bit? The msbuildArchitecture being x64 is a bit confusing here. Is the **System.Environment.Is64BitProcess** property within unit tests equal to **false** when they are executing using MSBuild?

Reply:
tested, Assert.True(System.Environment.Is64BitProcess) fails the test on x86 build